### PR TITLE
docs: add pruning section

### PR DESCRIPTION
**Pull request by tac0turtle**
- added a pruning section to joinmainnet section.

Signed-off-by: Marko Baricevic <marbar3778@yahoo.com>

Closes: #XXX

## Description

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/gaia/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/gaia/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

**Review by fedekunze — APPROVED** [review]

File: docs/gaia-tutorials/join-mainnet.md
Comment (fedekunze) [inline docs/gaia-tutorials/join-mainnet.md]:
```suggestion
3. `PruneSyncable`: This means that only the state of the last 100 and every 10,000th blocks will be saved.
```

File: docs/gaia-tutorials/join-mainnet.md
Comment (fedekunze) [inline docs/gaia-tutorials/join-mainnet.md]: If ...

**Review by fedekunze — CHANGES_REQUESTED** [review]
Please complete the `If...`

File: docs/gaia-tutorials/join-mainnet.md
Comment (tac0turtle) [reply docs/gaia-tutorials/join-mainnet.md]: not sure if was meant for here or somewhere else and I forgot to click into that window

**Review by alexanderbez — COMMENTED** [review]
I would wait until we have the pruning fix merged onto SDK `master` and we'll want to make note of the new changes to flushing to disk.

**Review by fedekunze — APPROVED** [review]

**Comment by tac0turtle** [comment]
I based this addition off of 0.37.6 of the sdk since it is what is used on the hub. Once the pruning is figured out in 0.38 i will make a section in the docs of the sdk with the new changes

**Pull request merged**
